pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -491,27 +541,59 @@ func (policy *regoEnforcer) enforce(enforcementPoint string, input map[string]in
 		return err
 	}
 
+	removeModule := false
+	if enforcementPoint == "load_fragment" {
+		if addModule, ok := results["add_module"].(bool); ok {
+			if !addModule {
+				removeModule = true
+			}
+		} else {
+			removeModule = true
+		}
+	}
+
 	if allowed {
 		err = policy.updateMetadata(results)
 		if err != nil {
 			return fmt.Errorf("unable to update metadata: %w", err)
 		}
-		return nil
+	} else {
+		err = policy.getReasonNotAllowed(enforcementPoint, input)
+	}
+
+	if removeModule {
+		delete(policy.modules, moduleID(input["issuer"].(string), input["feed"].(string)))
+		if compileError := policy.compile(); compileError != nil {
+			return fmt.Errorf("post rule error: %v, was unable to re-compile module: %v", err, compileError)
+		}
+	}
+
+	return err
+}
+
+func errorString(errors interface{}) string {
+	errorArray := errors.([]interface{})
+	output := make([]string, len(errorArray))
+	for i, err := range errorArray {
+		output[i] = fmt.Sprintf("%v", err)
 	}
+	return strings.Join(output, ",")
```

Comment:
I originally had it with newlines but @anmaxvl had indicated that it was preferred to keep errors on a single line. I'm very happy to change it to whatever is best for users.

Reply:
An example error I've seen in Azure Portal/Kusto that I find hard to read: 

Error: Failed to start container container_name, Error response: to create containerd task: failed to mount container storage: guest modify: guest RPC failure: layerPaths '[/run/mounts/m65 /run/mounts/m66 /run/mounts/m67 /run/mounts/m68 /run/mounts/m69 /run/mounts/m70 /run/mounts/m71 /run/mounts/m72 /run/mounts/m73 /run/mounts/m74 /run/mounts/m75 /run/mounts/m76 /run/mounts/m77 /run/mounts/m78 /run/mounts/m79 /run/mounts/m80 /run/mounts/m81 /run/mounts/m82 /run/mounts/m83]' already used in maximum number of container overlays: unknown

Not sure if your securitypolicyenforcer errors will display in Azure Portal, but if they look like this, it could be hard to figure out the root error.

Reply:
new lines aren't used in Go error messages.